contracts/MockAcrossMessageContract.sol

```diff
@@ -0,0 +1,40 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+// This interface is expected to be implemented by any contract
+// that expects to receive messages from the SpokePool.
+// Retrieved from https://github.com/across-protocol/contracts-v2/blob/master/contracts/SpokePool.sol
+interface AcrossMessageHandler {
+  function handleAcrossMessage(
+    address tokenSent,
+    uint256 amount,
+    bool fillCompleted,
+    address relayer,
+    bytes memory message
+  ) external;
+}
+
+contract MockAcrossMessageContract is AcrossMessageHandler {
+  function handleAcrossMessage(
+    address tokenSent,
+    uint256 amount,
+    bool fillCompleted,
+    address relayer,
+    bytes memory message
+  ) external view override {
+    // "Use" the variables to avoid a compilation warning.
+    tokenSent;
+    amount;
+    fillCompleted;
+    relayer;
+    message;
+    // For the case that we want to test a revert.
+    // We consider a message with a single byte 0x02 to be a revert.
+    require(!(message.length == 1 && message[0] == 0x02), "MockAcrossMessageContract: revert");
+    // Iterate from 0 to 1000 to simulate a long-running operation.
```

Comment:
is this long running? How much gas does this expect to spend? Since we seem to want to test that a message adds a non-zero amount of gas, why not just do somethingn even simpler like incrementing the loop from 0 --> 100 and doing nothing. The increment of the memory variable `i` is sufficient so I think removing the extra `address(this).balance` is kind of overkill

Reply:
Even calling into this function incurs a fixed cost so we don't even need the loop arguably

Reply:
The loop exists so that there is an easily viewed reason that we are expending gas outside the intrinsic gas costs of calling any function.

I wanted this function to touch the chain in some way so that it could be marked as `view` and not `virtual` to simulate what a realistic function could look like.

I've removed the `address(this).balance`, but I'd recommend we keep the look so that this function showcases that it's meant to expend more gas.